### ISSUE TITLE
Kubelet spams 'Failed to create existing container'  log messages.

### ISSUE THREAD

**Issue by Dorsosiv** [kubernetes/kubernetes#109339]
### What happened?

Hey, we are getting the following error messages when Kubelet is working as usual. 
```
Apr 06 14:54:33 deb10.eu.sosiv.io kubelet[2626199]: E0406 14:54:33.801083 2626199 manager.go:1123] Failed to create existing container: /pids/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-podadeea517_2589_41b3_88af_02c3131a0b1a.slice/crio-23c21ea8a10999235e300ba892c6b77c7bfc6e768cb6e483e498f0e93ded16ca.scope: Error finding container 23c21ea8a10999235e300ba892c6b77c7bfc6e768cb6e483e498f0e93ded16ca: Status 404 returned error &{%!s(*http.body=&{0xc0012fb6f8 <nil> <nil> false false {0 0} false false false <nil>}) {%!s(int32=0) %!s(uint32=0)} %!s(bool=false) <nil> %!s(func(error) error=0x83f4e0) %!s(func() error=0x83f5e0)}
Apr 06 14:54:33 deb10.eu.sosiv.io kubelet[2626199]: E0406 14:54:33.801594 2626199 manager.go:1123] Failed to create existing container: /pids/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod67dd1796_0c02_443b_8319_54401b20ecc3.slice/crio-816a633831f3e14d9c666d050ea6895034d29833827612b2e5cddc166f82b994.scope: Error finding container 816a633831f3e14d9c666d050ea6895034d29833827612b2e5cddc166f82b994: Status 404 returned error &{%!s(*http.body=&{0xc001f1d0e0 <nil> <nil> false false {0 0} false false false <nil>}) {%!s(int32=0) %!s(uint32=0)} %!s(bool=false) <nil> %!s(func(error) error=0x83f4e0) %!s(func() error=0x83f5e0)}
Apr 06 14:54:33 deb10.eu.sosiv.io kubelet[2626199]: E0406 14:54:33.802004 2626199 manager.go:1123] Failed to create existing container: /pids/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-podb7b3cabc_5227_4c47_b82f_21077a4d3064.slice/crio-2318c4bc814247675d78d92cb2838a78a505d409b3937ac71e4d3d4ffa228dd4.scope: Error finding container 2318c4bc814247675d78d92cb2838a78a505d409b3937ac71e4d3d4ffa228dd4: Status 404 returned error &{%!s(*http.body=&{0xc0012fb770 <nil> <nil> false false {0 0} false false false <nil>}) {%!s(int32=0) %!s(uint32=0)} %!s(bool=false) <nil> %!s(func(error) error=0x83f4e0) %!s(func() error=0x83f5e0)}

```  
I tried to see why the kubelet is trying to reach the folder called `/pids` which does not exist on our worker nodes. I did not find a reason in the config.yaml of Kubelet about that.

I tried to find a path similar to the `/pids/.....`  paths in the logs. I found '/sys/fs/cgroup/pids/kubepods.slice/kubepods-burstable.slice/' .  Which has the same path and under it, I did see the rest of the path that appears in the logs.  

Besides that, I did see that under the folder: `/sys/fs/cgroup/pids/kubepods.slice/kubepods-burstable.slice/` there are over 1000 directories of pods and containers.  Most of which are dead and no longer exist. 

I tried to compare to other clusters I have (1.21 Kubernetes and Openshift4 version 4.7). In both I saw that the number of directories that exists under `/sys/fs/cgroup/pids/kubepods.slice/kubepods-burstable.slice/` is less or equal to the amount of pods that were running at the same time on the worker nodes I checked in each cluster. 

### What did you expect to happen?

Kubelet is not finding directories because it searches for them in wrong path.  
Kubelet does not remove dead pods/containers slices directories from `/sys/fs/cgroup/pids/kubepods.slice/kubepods-burstable.slice/`

### How can we reproduce it (as minimally and precisely as possible)?

Create a Cluster of Kubernetes with Kubeadm with the same version of Kubernetes and crio engine on Debian machines with the version specified below (OS Version section of this issue).  
We are experiencing this behavior in another Kubernetes cluster. that was installed with Kubeadm and is running on Debian servers (with same version as the cluster for which the issue is created) with the same crio engine version.  

### Anything else we need to know?

I suspect that my issue might relate to this issue : https://github.com/kubernetes/kubernetes/issues/106957

### Kubernetes version

<details>

```bash 
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.0", GitCommit:"af46c47ce925f4c4ad5cc8d1fca46c7b77d13b38", GitTreeState:"clean", BuildDate:"2020-12-08T17:59:43Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.3", GitCommit:"816c97ab8cff8a1c72eccca1026f7820e93e0d25", GitTreeState:"clean", BuildDate:"2022-01-25T21:19:12Z", GoVersion:"go1.17.6", Compiler:"gc", Platform:"linux/amd64"}

```

</details>

### Cloud provider

<details>
None
</details>

### OS version

<details>

```bash
# On Linux:
$  cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

$ uname -a
Linux <server-name> 5.10.0-13-amd64 #1 SMP Debian 5.10.106-1 (2022-03-17) x86_64 GNU/Linux

```

</details>

### Install tools

<details>
kubeadm 
```bash
$ kubeadm version
kubeadm version: &version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.0", GitCommit:"ab69524f795c42094a6630298ff53f3c3ebab7f4", GitTreeState:"clean", BuildDate:"2021-12-07T18:15:11Z", GoVersion:"go1.17.3", Compiler:"gc", Platform:"linux/amd64"}
```
</details>

### Container runtime (CRI) and version (if applicable)

<details>

```
Version:  0.1.0  
RuntimeName:  cri-o  
RuntimeVersion:  1.22.1  
RuntimeApiVersion:  v1alpha2  
```

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by Dorsosiv** [comment]
Hey, Ill answer shortly.

**Comment by Dorsosiv** [comment]
* Kubeadm and Kubelet versions are like the server (1.23.0).
```bash 
$ kubeadm version
kubeadm version: &version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.0", GitCommit:"ab69524f795c42094a6630298ff53f3c3ebab7f4", GitTreeState:"clean", BuildDate:"2021-12-07T18:15:11Z", GoVersion:"go1.17.3", Compiler:"gc", Platform:"linux/amd64"}
```

```bash
$ kubelet --version
Kubernetes v1.23.0
```

* The installation was a clean installation. This is not happenning because of an upgrade. And we are not doing an upgrade to the cluster

* There are no additional steps to perform.

* The snippet of logs that I have sent, is a repeating itself for tons of pods that do not exist anymore.  
Im not allowed to send full Kubelet logs from server.
So I took all messages that appeared in the logs of the past hour and I removed all repetitions and kept the order as it was.
I also added some notes about the logs. 

[logs-for-github.txt](https://github.com/kubernetes/kubernetes/files/8441315/logs-for-github.txt)
